app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -295,21 +318,18 @@ function shouldReverseScale(scatterColor) {
 
 /** get the array of plotly traces for plotting */
 export function getPlotlyTraces({
-  axes,
-  data,
-  annotType,
-  annotName,
   genes,
   isAnnotatedScatter,
   isCorrelatedScatter,
   scatterColor,
-  dataScatterColor,
-  pointAlpha,
-  pointSize,
-  is3D,
   hiddenTraces,
   scatter
 }) {
+  const { axes, data, pointAlpha, pointSize, is3D } = scatter
+  const dataScatterColor = scatter.scatterColor
+  const annotName = scatter.annotParams.name
+  const annotType = scatter.annotParams.type
```

Comment:
It's nice that now the arguments to this function are all clearly independent state objects--that makes life easier for the caller, but at the expense of more of our code exposed to the `scatter` 'god object'.  I wonder if the best of both worlds is moving the deconstruction to the function parameters like: 
```
function getPlotlyTraces({
  genes, 
  isAnnotatedScatter, 
  scatter: {axes, data, pointAlpha, pointSize, is3D, 
    scatterColor: dataScatterColor, 
    annotParams: {name: annotName, type: annotType}
  }
})```

Reply:
I didn't know this was possible!  The syntax is a bit complex and dense, but seems like an excellent compromise.  Done in de38b69.